library/core/src/enums.rs

```diff
@@ -0,0 +1,58 @@
+//! For introspecting and converting between fieldless enums and numbers.
+
+/// Converts an enum to its underlying discriminant.
+///
+/// This trait is automatically implemented for all fieldless enum types which have an explicit
+/// numeric repr. It may not be manually implemented.
+#[unstable(feature = "enum_as_repr", issue = "86772")]
+#[cfg_attr(not(bootstrap), lang = "AsRepr")]
+pub unsafe trait AsRepr: Sized {
```

Comment:
It's unclear to me whether this trait should still be `unsafe` if it's only implementable by the language itself. I guess it doesn't _really_ matter either way?

Reply:
I guess it is a nice safeguard if it accidentally becomes possible to manually implement it anyway.